### PR TITLE
feat(ontology): #2383 sunburst for phenotypes ontology

### PR DESCRIPTION
**Pull request by francisl**
tests on the way

### CODE REVIEW

File: src/components/UserDashboard/Ontology/index.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    const Header = <CardHeader title="Ontology" />;
+    const loading = false;
+
+    return (
+      <DashboardCard Header={Header} inactive={loading}>
+        {loading ? (
+          <ChartContentSpinner />
+        ) : (
+          <div className="card-content-center">
+            {/* <div id="D3Container" ref={this.d3ContainerRef} /> */}
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/Ontology/index.tsx:53]: is this comment needed?

File: src/components/UserDashboard/Ontology/index.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+class OntologySunburst extends React.Component<OntologySunburstProps, OntologySunburstState> {
+  state = {
+    data: null,
+  };
+
+  componentDidMount() {
+    const ontologyStore = new PhenotypeStore();
+    ontologyStore
+      .fetch()
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/Ontology/index.tsx:29]: I think async/await looks better but it's a personal taste...why do you think of it?

File: src/components/UserDashboard/Ontology/index.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      .then(() => {
+        const data = ontologyStore.getTree();
+        console.log('data : ', data);
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/Ontology/index.tsx:32]: unneeded console.log

File: src/components/UserDashboard/Ontology/index.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  render() {
+    const { data } = this.state;
+    const Header = <CardHeader title="Ontology" />;
+    const loading = false;
+
+    return (
+      <DashboardCard Header={Header} inactive={loading}>
+        {loading ? (
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/Ontology/index.tsx:49]: always false???

File: src/components/UserDashboard/index.js [hunk 5]
```diff
@@ -49,10 +50,13 @@ export default compose(
     <div className={userDashboardContent}>
       <h1 className={dashboardTitle}>My Dashboard</h1>
       {/* [NEXT] SizeProvider here is the only usage of 'react-sizeme' */}
-      <Row gutter={[30, 30]} align={'top'} style={{paddingLeft: 15, paddingRight: 15, top:-15}}>
+      <Row gutter={[30, 30]} align={'top'} style={{ paddingLeft: 15, paddingRight: 15, top: -15 }}>
```
Comment (evans-g-crsj) [inline src/components/UserDashboard/index.js:53]: if simple, I d suggest using css

File: src/components/Charts/Sunburst/sunburst-d3.jsx [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+  const arc = d3
+    .arc()
+    .startAngle((d) => d.x0)
+    .endAngle((d) => d.x1)
+    .padAngle((d) => Math.min((d.x1 - d.x0) / 1.2, 0.05))
+    .padRadius(radius)
+    .innerRadius((d) => (d.y1 <= 2 ? d.y0 * (radius + 28) : d.y0 * (radius + 2)))
+    .outerRadius((d) => Math.max(d.y0 * radius, d.y1 * radius - 2));
+
+  const partition = (data) => {
```
[the remark this answers is not in the document]
Comment (francisl) [reply src/components/Charts/Sunburst/sunburst-d3.jsx:21]: I think we need at least one by default, and this one is quite generic. I'll say we should make it configurable.

File: src/components/OntologyBrowser/store.ts [hunk 7]
```diff
@@ -96,6 +104,23 @@ export class PhenotypeStore {
     return nKeys;
   };
 
+  getTree = (maxDepth: number = 2) => {
+    const newTree = [...this.tree];
+    if (newTree.length === 0) return [];
```
[the remark this answers is not in the document]
Comment (francisl) [reply src/components/OntologyBrowser/store.ts:109]: good point

File: src/components/UserDashboard/Ontology/index.tsx (hunk 1 above)
Comment (francisl) [reply src/components/UserDashboard/Ontology/index.tsx:53]: it was already clean in my next iteration, will be fix with the update

File: src/components/UserDashboard/Ontology/index.tsx (hunk 3 above)
Comment (francisl) [reply src/components/UserDashboard/Ontology/index.tsx:32]: already fix, wait for update PR

File: src/components/UserDashboard/Ontology/index.tsx (hunk 4 above)
Comment (francisl) [reply src/components/UserDashboard/Ontology/index.tsx:49]: should not, fixed

File: src/components/UserDashboard/index.js (hunk 5 above)
Comment (francisl) [reply src/components/UserDashboard/index.js:53]: didn't catch that since I didn't touch that part, it was auto cleaned by the linter, will fix

**Pull request merged**